lib/src/Uncertainty/Distribution/TruncatedDistribution.cxx

```diff
@@ -135,37 +132,30 @@ TruncatedDistribution::TruncatedDistribution(const Distribution & distribution,
     const Interval & truncationInterval,
     const Scalar thresholdRealization)
   : DistributionImplementation()
+  , bounds_(truncationInterval)
 {
   setName("TruncatedDistribution");
-  if (distribution.getDimension() != 1) throw InvalidArgumentException(HERE) << "Error: can truncate only distribution with dimension=1, here dimension=" << distribution.getDimension();
   distribution_ = distribution;
   setThresholdRealization(thresholdRealization);
-  setDimension(1);
+  setDimension(distribution.getDimension());
   setDescription(distribution.getDescription());
-  lowerBound_ = truncationInterval.getLowerBound()[0];
-  upperBound_ = truncationInterval.getUpperBound()[0];
-  finiteLowerBound_ = truncationInterval.getFiniteLowerBound()[0] == 1;
-  finiteUpperBound_ = truncationInterval.getFiniteUpperBound()[0] == 1;
-  cdfLowerBound_ = distribution.computeCDF(Point(1, lowerBound_));
-  cdfUpperBound_ = distribution.computeCDF(Point(1, upperBound_));
+  cdfLowerBound_ = distribution.computeCDF(bounds_.getLowerBound());
+  cdfUpperBound_ = distribution.computeCDF(bounds_.getUpperBound());
   if (cdfLowerBound_ >= cdfUpperBound_) throw InvalidArgumentException(HERE) << "Error: the truncation interval does not contain a non-empty part of the support of the distribution";
-  pdfLowerBound_ = distribution.computePDF(Point(1, lowerBound_));
-  pdfUpperBound_ = distribution.computePDF(Point(1, upperBound_));
+  pdfLowerBound_ = distribution.computePDF(bounds_.getLowerBound());
+  pdfUpperBound_ = distribution.computePDF(bounds_.getUpperBound());
   normalizationFactor_ = 1.0 / (cdfUpperBound_ - cdfLowerBound_);
```

Comment:
The normalization factor is the inverse of the probability of the truncation interval, so the formula is wrong for multivariate distributions